From 23b9bf718a68723e7a0e59676a19af1a42bbf20b Mon Sep 17 00:00:00 2001
From: Haris Okanovic <haris.okanovic@ni.com>
Date: Mon, 25 Nov 2019 12:49:07 -0600
Subject: [PATCH 1/3] opkg.py/get_file_list_dir(): Decode subprocess string,
 python3 support

String split fails on newline (\n) chars in python3 unless subprocess
output string is decoded as text. This results in fn containing a bad
filename with trailing newline (\n) char, which subsequently breaks
open() in get_file_list().

Tested `opkg-make-index -l Packages.filelist ...` in python 2 and 3.

Signed-off-by: Haris Okanovic <haris.okanovic@ni.com>

Upstream-Status: Submitted
---
 opkg.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/opkg.py b/opkg.py
index 43a2ead..ddb2c38 100644
--- a/opkg.py
+++ b/opkg.py
@@ -384,6 +384,7 @@ class Package(object):
                 error = subprocess.CalledProcessError(retcode, cmd)
                 error.output = output
                 raise error
+            output = output.decode("utf-8")
             return output
 
         if not self.fn:
-- 
2.24.0

